File: src/main/java/com/bombproj/framework/SessionConfig.java
package com.bombproj.framework;

import com.bombproj.constants.UserType;
import lombok.Data;

import java.io.Serializable;

@Data
public class SessionConfig implements Serializable {

    private static final long serialVersionUID = 1L;

    public static final String KEY_VERIFICATION_CODE = "__ver_code";
    public static final String KEY_SESSION_CONFIG = "__session_config";

    private String userId;
    private String userName;
    private String nickName;
    private UserType userType;

    private static final SessionConfig TESTER = new SessionConfig();
    static {
        TESTER.setUserId("admin");
        TESTER.setUserName("admin");
        TESTER.setNickName("admin");
        TESTER.setUserType(UserType.NORMAL);
    }

    public static SessionConfig current() {
        return TESTER;
//        return (SessionConfig) WebUtils.getSessionAttribute(KEY_SESSION_CONFIG);
    }

}

File: src/main/scala/com/bombproj/framework/SessionConfig.scala
package com.bombproj.framework

import com.bombproj.constants.UserType

@SerialVersionUID(1L)
case class SessionConfig() {

    var userId: String = _
    var userName: String = _
    var nickName: String = _
    var userType: UserType = _

}

object SessionConfig {

    val KEY_VERIFICATION_CODE = "__ver_code"
    val KEY_SESSION_CONFIG = "__session_config"

    private val currentUser = new SessionConfig()
    currentUser.userId = "admin"
    currentUser.userName = "admin"
    currentUser.nickName = "admin"
    currentUser.userType = UserType.NORMAL

    def current(): SessionConfig = {
        currentUser
//        WebUtils.getSessionAttribute(KEY_SESSION_CONFIG).asInstanceOf[SessionConfig]
    }

}
